# 2.3: Анализ рынка недвижимости
Ответьте на несколько вопросов:
- В каком диапазоне изменяются стоимости недвижимости?
- Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
- Как много домов с разными этажами в данных?
- Насколько хорошие состояния у домов в данных?
- Найдите года, когда построили первый дом, когда построили последний дом в данных?

### Часть 1

In [20]:
import pandas as pd

In [21]:
df = pd.read_csv('kc_house_data.csv')

In [22]:
df.head(2)

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   

   sqft_lot15  
0        5650  
1        7639  

[2 rows x 21 columns]

Наименования столбцов и их суть:
- id - Уникальный ID для каждого дома
- date - Дата продажи дома
- price - Стоимость продажи дома
- bedrooms - Кол-во спален
- bathrooms - Кол-во ванных комнат (0.5 - туалет без душа)
- sqft_living - Кв. метры жилые
- sqft_lot - Кв. метры общие
- floors - Кол-во этажей
- waterfront - Есть набержная или нет
- view - Значение от 0 до 4 насколько хороший вид
- condition - Значение от 1 до 5 насколько хорошее состояние
- grade - Значение от 1 до 13, где 1-3 плохая конструкция здания и дизайн, 7 - средний уровень конструкции и дизайна, 11-13 - высокое качество конструкции и дизайна
- sqft_above - Кв. метры дома, которые находятся выше земли
- sqft_basement - Кв. метры дома, которые находятся ниже земли
- yr_built - Год постройки дома
- yr_renovated - Год ремонта дома
- zipcode - Индекс
- lat - Широтма
- long - Долгота
- sqft_living15 - Кв. метры жилой площади у 15 соседей
- sqft_lot15 - Кв. метры общей площади у 15 соседей

In [23]:
df['price'].min()

75000.0

In [24]:
df['price'].max()

7700000.0

Вывод: диапазон цен на дома составляют от 75 000 до 7 700 000

### Часть 2

In [25]:
sqft_living_mean = df['sqft_living'].mean()
sqft_living_mean

2079.8997362698374

In [26]:
sqft_lot_mean = df['sqft_lot'].mean()
sqft_lot_mean

15106.967565816869

In [27]:
part = sqft_living_mean/sqft_lot_mean
part

0.1376781757959227

Вывод:  0,14 - доля, которую в среднем занимают жилая площадь от всей площади по всем домам

### Часть 3

In [28]:
df['floors'].describe()

count    21613.000000
mean         1.494309
std          0.539989
min          1.000000
25%          1.000000
50%          1.500000
75%          2.000000
max          3.500000
Name: floors, dtype: float64

In [37]:
df['floors'].unique()

array([1. , 2. , 1.5, 3. , 2.5, 3.5])

In [38]:
one_floors = df[df['floors'] == 1]['id'].count()
one_floors

10680

In [42]:
one_floors_5 = df[df['floors'] == 1.5]['id'].count()
one_floors_5

1910

In [39]:
two_floors = df[df['floors'] == 2]['id'].count()
two_floors

8241

In [43]:
two_floors_5 = df[df['floors'] == 2.5]['id'].count()
two_floors_5

161

In [40]:
three_floors = df[df['floors'] == 3]['id'].count()
three_floors

613

In [44]:
three_floors_5 = df[df['floors'] == 3.5]['id'].count()
three_floors_5

8

In [41]:
four_floors = df[df['floors'] == 4]['id'].count()
four_floors

0

In [45]:
sum_house_floors = one_floors + one_floors_5 + two_floors + two_floors_5 + three_floors + three_floors_5 + four_floors
sum_house_floors

21613

Вывод: в предложенной выборке из 21 613 зданий есть:
- 1 этаж: 10680 зданий 
- 1,5 этажа: 1910 зданий 
- 2 этажа: 8241 здание
- 2,5 этажа: 161 здание 
- 3 этажа: 613 зданий 
- 3,5 этажа: 8 зданий 

### Часть 4

In [47]:
df['condition'].unique()

array([3, 5, 4, 1, 2])

In [48]:
df['condition'].mean()

3.4094295100171195

In [52]:
condition_1 = df[df['condition'] == 1]['id'].count()
condition_1

30

In [53]:
condition_2 = df[df['condition'] == 2]['id'].count()
condition_2

172

In [54]:
condition_3 = df[df['condition'] == 3]['id'].count()
condition_3

14031

In [55]:
condition_4 = df[df['condition'] == 4]['id'].count()
condition_4

5679

In [56]:
condition_5 = df[df['condition'] == 5]['id'].count()
condition_5

1701

In [57]:
condition_summa = condition_1 + condition_2 + condition_3 + condition_4 + condition_5
condition_summa

21613

Вывод: средняя оценка состояния домой в выборке: 3,41

### Часть 5

In [58]:
df['yr_built'].min()

1900

In [59]:
df['yr_built'].max()

2015

Вывод: первый дом построили в 1900 году, а последний дом построили в 2015 году